server/bleep/src/webserver/answer.rs

```diff
@@ -129,11 +134,36 @@ pub async fn handle(
     // reorder snippets
     snippets.swap(relevant_snippet_index, 0);
 
+    if let Some(ref segment) = *app.segment {
```

Comment:
Can we move this logic into the `Segment` struct like `segment.send(segment::Track { ... }).await`, and the struct into a new file? Would be a bit cleaner, and contains all the relevant logic.